cpp/ql/src/semmle/code/cpp/exprs/Cast.qll

```diff
@@ -519,22 +638,18 @@ class UuidofOperator extends Expr, @uuidof {
 }
 
 /**
- * A C++ `typeid` expression which provides runtime type information
- * about an expression or type.
+ * A C++ `typeid` expression which provides run-time type information (RTTI)
+ * about its argument.
+ *
+ * Please see https://en.cppreference.com/w/cpp/language/typeid for more
+ * information.
+ * ```
+ * Base *ptr = new Derived;
+ * const std::type_info &info1 = typeid(ptr);
+ * printf("the type of ptr is: %s\n", typeid(ptr).name);
+ * ```
  */
 class TypeidOperator extends Expr, @type_id {
-  /**
-   * Gets the type that is returned by this typeid expression.
-   *
-   * For example in the following code the `typeid` returns the
-   * type `MyClass *`.
-   *
-   * ```
-   * MyClass *ptr;
-   *
-   * printf("the type of ptr is: %s\n", typeid(ptr).name);
-   * ```
-   */
```

Comment:
Why has this QLDoc comment been deleted.

Reply:
Because it was in the wrong place (inside the **TypeidOperator** definition instead of above it).  The description is also incomplete in that it does not mention the run-time polymorphism that is involved.

Reply:
OK.